src/sync/update_tx.js

```diff
@@ -1,157 +1,214 @@
+    tx.gasUsed = resp[0].gasUsed;
+    tx.blockNum = resp[0].blockNumber;
+  }
+}
+
+// Update the DB with new Transaction info
+async function updateDB(tx, db) {
+  if (tx.status !== 'PENDING') {
     try {
-      tx = await db.Transactions.update(
-        { address: approveTxid },
+      logger.debug(`Update: Transaction ${tx.type} txid:${tx._id}`);
+      const updateRes = await db.Transactions.update(
+        { _id: tx._id },
         {
           $set: {
-            status: txReceipt.status,
-            gasUsed: txReceipt.gasUsed,
-            blockNum: txReceipt.blockNum,
+            status: tx.status,
+            gasUsed: tx.gasUsed,
+            blockNum: tx.blockNum,
           },
-        }, {},
+        }, 
+        {
+          returnUpdatedDocs: true,
+        },
       );
+      const updatedTx = updateRes[1];
+
+      // Execute follow up tx
+      if (updatedTx && updatedTx.status === 'SUCCESS') {
+        await executeFollowUpTx(updatedTx, db);
+      }
     } catch (err) {
-      logger.error(`Error updateApproveTx ${approveTxid}: ${err.message}`);
+      logger.error(`Error: Update Transaction ${tx.type} txid:${tx._id}: ${err.message}`);
       throw err;
     }
   }
-
-  if (txReceipt.status === 'SUCCESS' && tx.type.startsWith('APPROVE')) {
-    updateApprovedTx(tx, db);
-  }
 }
 
-async function updateApprovedTx(approveTx, db) {
+// Execute follow-up transaction
+async function executeFollowUpTx(tx, db) {
   const Transactions = db.Transactions;
-  if (approveTx.type === 'APPROVESETRESULT') {
-    let setResultTxid;
-    try {
-      const resp = await fetch('http://localhost:5555/set-result', {
-        method: 'POST',
-        body: JSON.stringify({
-          contractAddress: approveTx.entityId,
-          resultIndex: approveTx.optionIdx,
-          senderAddress: approveTx.senderAddress,
-        }),
-        headers: { 'Content-Type': 'application/json' },
-      }).then(res => res.json());
-      setResultTxid = resp.result.txid;
-    } catch (err) {
-      logger.error(`Error call /set-result: ${err.message}`);
-      throw err;
+  let txid;
+  switch (tx.type) {
+    // Approve was reset to 0. Sending approve for consensusThreshold.
+    case 'RESETAPPROVESETRESULT': {
+      try {
+        const approveTx = await bodhiToken.approve({
+          spender: tx.topicAddress,
+          value: tx.amount,
+          senderAddress: tx.senderAddress,
+        });
+        txid = approveTx.txid;
+      } catch (err) {
+        logger.error(`Error calling BodhiToken.approve: ${err.message}`);
+        throw err;
+      }
+
+      await DBHelper.insertTransaction(Transactions, {
+        _id: txid,
+        txid,
+        version: tx.version,
+        type: 'APPROVESETRESULT',
+        status: 'PENDING',
+        senderAddress: tx.senderAddress,
+        topicAddress: tx.topicAddress,
+        oracleAddress: tx.oracleAddress,
+        optionIdx: tx.optionIdx,
+        token: 'BOT',
+        amount: tx.amount,
```

Comment:
convert amount from `decimalstring` to `hexstring`

Reply:
will leave as decimal string as discussed.

Reply:
same note as above